src/azure-cli/azure/cli/command_modules/containerapp/custom.py

```diff
@@ -2916,6 +2916,8 @@ def set_registry(cmd, name, resource_group_name, server, username=None, password
     _validate_subscription_registered(cmd, CONTAINER_APPS_RP)
     if (username or password) and identity:
         raise MutuallyExclusiveArgumentError("Use either identity or username/password.")
+    if identity and identity.lower() != "system" and not is_valid_resource_id(identity):
```

Comment:
why not validate in RP

Reply:
Currently the response from request is :
`The following field(s) are either invalid or missing. Field 'configuration.Registries<acr>.azurecr.io.Identity' is invalid with details: 'Invalid value: "mic-acr-puller": Managed Identity does not exist';..`

I think we can add validation in CLI.